nio/service/tests/test_service_base.py

```diff
@@ -241,17 +241,34 @@ def test_notify_management_signal(self):
 
     def test_failed_start(self):
         """Test service start failure"""
-        context = ServiceContext({"id": "TestFailedStart"},
-                                 [],
-                                 BlockRouter)
+        class Block1(Block):
+            pass
+
+
+        class Block2(Block):
+            pass
+
+
+        class TestException(Exception):
+            def __init__(self):
+                super().__init__("bad things")
 
         service = Service()
-        service.configure(context)
+        Block1.start = Mock()
+        Block2.start = Mock(side_effect=TestException)
+        blocks = [{"type": Block1,
+                   "properties": {"id": "block1"}},
+                  {"type": Block2,
+                   "properties": {"id": "block2"}}]
 
-        service.start = Mock(side_effect=Exception())
-        with self.assertRaises(Exception):
+        service.do_configure(ServiceContext(
+            properties={"id": "ServiceId"},
+            blocks=blocks,
+            block_router_type=BlockRouter,
+        ))
+        try:
```

Comment:
We want to actually make sure this is raising the exception here. By including the assertion in the except block we won't actually check anything if it doesn't raise that exception. Instead we should use the `assertRaises` to check it:
```python
with self.assertRaises(BlockException) as e:
    service.do_start()
self.assertEqual(e.label, "block2")
```

Reply:
:+1: